07_errors/n0npax/leveldbStore.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+var levelDBPath = "/tmp/leveldb"
+
+// NewLevelDBStorer creates new storer for leveldb
+func NewLevelDBStorer() *LevelDBStore {
+	db, err := leveldb.OpenFile(levelDBPath, nil)
+	dbErrorPanic(err)
+	return &LevelDBStore{nextID: 0, ldb: db}
+}
+
+// CreatePuppy creates puppy
+func (l *LevelDBStore) CreatePuppy(puppy *Puppy) (int, error) {
+	l.Lock()
+	defer l.Unlock()
+	id, err := l.putPuppy(l.nextID, puppy)
+	l.nextID++
+	return id, err
+}
+
+// ReadPuppy reads puppy from backend
+func (l *LevelDBStore) ReadPuppy(id int) (*Puppy, error) {
+	byteID := []byte(strconv.Itoa(id))
+	if puppyData, err := l.ldb.Get(byteID, nil); err == nil {
+		var puppy Puppy
+		err := json.Unmarshal(puppyData, &puppy)
+		if err != nil {
+			return nil, Errorf(ErrCodeInternal, "Internal error. Could not cast stored data to puppy object")
+		}
+		return &puppy, nil
+	}
+	return nil, Errorf(ErrCodeNotFound, fmt.Sprintf("Puppy with ID (%v) not found", id))
+}
+
+// UpdatePuppy updates puppy
+func (l *LevelDBStore) UpdatePuppy(id int, puppy *Puppy) error {
+	if id != puppy.ID {
+		return Errorf(ErrCodeInvalidInput, "ID is corrupted. Please ensure object ID matched provided ID")
+	}
+	l.Lock()
+	defer l.Unlock()
+	if _, err := l.ReadPuppy(id); err != nil {
+		return Errorf(ErrCodeNotFound, fmt.Sprintf("Puppy with ID (%v) not found", id))
+	}
+	_, err := l.putPuppy(id, puppy)
+	return err
+}
+
+// DeletePuppy deletes puppy
+func (l *LevelDBStore) DeletePuppy(id int) (bool, error) {
+	l.Lock()
+	defer l.Unlock()
+	if _, err := l.ReadPuppy(id); err != nil {
+		return false, Errorf(ErrCodeNotFound, fmt.Sprintf("Puppy with ID (%v) not found", id))
```

Comment:
`Sprintf` inside `Errorf` is unnecessary.

Reply:
fixed